woocommerce-gateway-payu.php

```diff
@@ -180,6 +180,162 @@ function enqueue_payu_admin_assets()
     add_action('admin_enqueue_scripts', 'enqueue_payu_admin_assets');
 }
 
+function get_installment_option($option)
+{
+
+    $paymentGateways = WC()->payment_gateways->payment_gateways();
+    $result = null;
+
+    if (array_key_exists('payuinstallments', $paymentGateways)) {
+
+        /**
+         * @var  $installmentsGateway WC_Gateway_PayuInstallments
+         */
+        $installmentsGateway = $paymentGateways['payuinstallments'];
+
+        switch ($option) {
+            case 'pos_id':
+                $result = $installmentsGateway->pos_id;
+                break;
+            case 'widget_key':
+                $result = $installmentsGateway->pos_widget_key;
+                break;
+        }
+    }
+
+    return $result;
+}
+
+
+if(get_option('woocommerce_payuinstallments_settings')['credit_widget_on_listings'] === 'yes') {
+    add_action('woocommerce_after_shop_loop_item', 'installments_mini');
+}
+
+if(get_option('woocommerce_payuinstallments_settings')['credit_widget_on_product_page'] === 'yes') {
+    add_action('woocommerce_before_add_to_cart_form', 'installments_mini');
+}
+
+function installments_mini() {
+    if(get_woocommerce_currency() !== 'PLN') {
+        return;
+    }
+    global $product;
+    $price = $product->get_price();
+    $productId = $product->get_id();
+
+    $posId = get_installment_option('pos_id');
+    $widgetKey = get_installment_option('widget_key');
+
+    wp_enqueue_script('payu-installments-widget', 'https://static.payu.com/res/v2/widget-mini-installments.js', ['jquery'], PAYU_PLUGIN_VERSION);
+
+    ?>
+        <div>
+            <p><span id="installment-mini-<?php echo esc_html($productId)?>"></span></p>
+            <script type="text/javascript">
+                (function ($) {
+                    $(document).ready(function(){
+                        var value = <?php echo esc_html($price)?>;
+                        var options = {
+                            creditAmount: value,
+                            posId: '<?php echo esc_html($posId)?>',
+                            key: '<?php echo esc_html($widgetKey)?>',
+                            showLongDescription: true
+                        };
+                        OpenPayU.Installments.miniInstallment('#installment-mini-<?php echo esc_html($productId)?>', options);
+                    });
+                })(jQuery);
+            </script>
+        </div>
+    <?php
+}
+
+
+if(get_option('woocommerce_payuinstallments_settings')['credit_widget_on_cart_page'] === 'yes') {
+    add_action('woocommerce_cart_totals_after_order_total', 'installments_mini_cart');
+}
+
+function installments_mini_cart() {
+    if(get_woocommerce_currency() !== 'PLN') {
```

Comment:
Na stronie koszyka można wybrać już metodą wysyłki i jak wybierzemy taką, która nie zawiera rat to pokazuje się mniratka.

Reply:
done